homeassistant/components/aten_pe/switch.py

```diff
@@ -0,0 +1,114 @@
+"""The ATEN PE switch component."""
+
+import logging
+
+from atenpdu import AtenPE
+import voluptuous as vol
+
+from homeassistant.components.snmp.const import (
+    CONF_AUTH_KEY,
+    CONF_COMMUNITY,
+    CONF_PRIV_KEY,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+)
+from homeassistant.components.switch import (
+    DEVICE_CLASS_OUTLET,
+    PLATFORM_SCHEMA,
+    SwitchDevice,
+)
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_COMMUNITY = "private"
+DEFAULT_USERNAME = "administrator"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_COMMUNITY, default=DEFAULT_COMMUNITY): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_AUTH_KEY): cv.string,
+        vol.Optional(CONF_PRIV_KEY): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the ATEN PE switch."""
+    dev = AtenPE(
+        node=config.get(CONF_HOST),
```

Comment:
Voluptuous handles the defaults/requirements, therefore, you can just use `config[CONF_HOST]`

Reply:
I left it unchanged for optional elements without defaults. Is that how it's supposed to be?